tendrl/node_agent/monitoring/collectd/collectors/gluster/low_weight/tendrl_glusterfs_brick_utilization.py

```diff
@@ -49,262 +51,57 @@ def __init__(self):
                 )
             self.etcd_client = etcd.Client(**_etcd_args)
 
-    def _get_mount_point(self, path):
-        mount = os.path.realpath(path)
-        while not os.path.ismount(mount):
-            mount = os.path.dirname(mount)
-        return mount
-
-    def _parse_proc_mounts(self, filter=True):
-        mount_points = {}
-        with open('/proc/mounts', 'r') as f:
-            for line in f:
-                if line.startswith("/") or not filter:
-                    mount = {}
-                    tokens = line.split()
-                    mount['device'] = tokens[0]
-                    mount['fsType'] = tokens[2]
-                    mount['mountOptions'] = tokens[3]
-                    mount_points[tokens[1]] = mount
-        return mount_points
-
-    def _get_stats(self, mount_point):
-        data = os.statvfs(mount_point)
-        total = (data.f_blocks * data.f_bsize)
-        free = (data.f_bfree * data.f_bsize)
-        used_percent = 100 - (100.0 * free / total)
-        total_inode = data.f_files
-        free_inode = data.f_ffree
-        used_percent_inode = 100 - (100.0 * free_inode / total_inode)
-        used = total - free
-        used_inode = total_inode - free_inode
-        return {'total': total,
-                'free': free,
-                'used_percent': used_percent,
-                'total_inode': total_inode,
-                'free_inode': free_inode,
-                'used_inode': used_inode,
-                'used': used,
-                'used_percent_inode': used_percent_inode}
-
-    def get_lvs(self):
-        _lvm_cmd = (
-            "lvm vgs --unquoted --noheading --nameprefixes "
-            "--separator $ --nosuffix --units m -o lv_uuid,"
-            "lv_name,data_percent,pool_lv,lv_attr,lv_size,"
-            "lv_path,lv_metadata_size,metadata_percent,vg_name"
-        )
+    def get_brick_utilization(self):
+        bricks_utilization = {}
         try:
-            cmd = Popen(
-                shlex.split(
-                    _lvm_cmd
-                ),
-                stdin=open(os.devnull, "r"),
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-                close_fds=True
-            )
-            stdout, stderr = cmd.communicate()
-            if stderr:
-                collectd.error(
-                    'Failed to fetch lvs. Error %s' % stderr
+            bricks = self.etcd_client.read(
```

Comment:
IMHO I wouldn't want to rely on etcd for anything...
The plugins in my view should have independent(As much as possible) existence immaterial of presence or absence tendrl..(If I remember correctly, there were plans of moving the plugins into a new repo under gluster org )...
For that matter, I am even ok with importing a tendrl module here(if replicating logic is duplication of efforts and considerably tedious) but relying on existence of functional tendrl stack(is somehow not convincing to me)
But its possible that things have changed now... So, my review would probably be stale now ;)

Reply:
yes, i agree with your comment, also this PR also will increase the number of etcd calls also,  This PR is just optional. for now we don't want to take this PR .

Reply:
Thanks for your review @anmolbabu :)

Reply:
Np :)